### PR TITLE
expose OUTGOING_WEBHOOK_TIMEOUT as env var

### PR DESCRIPTION
**Pull request by KevinDW-Fluxys**
# What this PR does
It adds functionality to be able to configure the outgoing webhook timeout from an environment variable.

## Which issue(s) this PR fixes
Running into timeouts when outgoing webhooks take longer than 4 seconds (which is exceptional, but can happen) the webhook reports failure, while it still might have succeeded on the webhook side.

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/webhooks/utils.py [hunk 1]
```diff
@@ -10,8 +10,9 @@
 from apps.labels.utils import get_alert_group_labels_dict, get_labels_dict, is_labels_feature_enabled
 from apps.schedules.ical_utils import list_users_to_notify_from_ical
 from common.jinja_templater import apply_jinja_template
+from common.utils import getenv_boolean
 
-OUTGOING_WEBHOOK_TIMEOUT = 4
+OUTGOING_WEBHOOK_TIMEOUT = getenv_boolean("OUTGOING_WEBHOOK_TIMEOUT", default=4)
```
[the remark this answers is not in the document]
Comment (KevinDW-Fluxys) [reply engine/apps/webhooks/utils.py]: Thanks for the review, I have updated the code accordingly. I wasn't sure if the reference in `engine/apps/alerts/utils.py` should receive the same treatment, but I added it for uniformity. If not needed let me know and I will remove it again.

File: docs/sources/outgoing-webhooks/_index.md [hunk 2]
```diff
@@ -111,7 +111,8 @@ If no integrations are selected the outgoing webhook will trigger for any integr
 
 The destination URL the outgoing webhook will make a request to. This must be a FQDN.
 
-> ⚠️ **Note** the destination server must respond back within 4 seconds or it will result in a timeout
+> ⚠️ **Note** the destination server must respond back within 4 seconds (by default) or it will result in a timeout
+> This timeout is configurable via the environment variable OUTGOING_WEBHOOK_TIMEOUT
```
Comment (joeyorlando) [inline docs/sources/outgoing-webhooks/_index.md:115]:
```suggestion
> For open source deployments, this timeout is configurable by setting the environment variable OUTGOING_WEBHOOK_TIMEOUT
```

File: engine/apps/webhooks/tests/test_webhook.py [hunk 3]
```diff
@@ -2,10 +2,9 @@
 
 import pytest
 from requests.auth import HTTPBasicAuth
-
+from apps.base.utils import live_settings
```
Comment (joeyorlando) [inline engine/apps/webhooks/tests/test_webhook.py:5]:
```suggestion
from django.conf import settings
```

File: engine/apps/webhooks/tests/test_webhook.py [hunk 4]
```diff
@@ -236,7 +235,7 @@ def test_make_request(make_organization, make_custom_webhook):
             webhook.make_request("url", {"foo": "bar"})
             expected_call = getattr(mock_requests, method.lower())
             assert expected_call.called
-            assert expected_call.call_args == call("url", timeout=OUTGOING_WEBHOOK_TIMEOUT, foo="bar")
+            assert expected_call.call_args == call("url", timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, foo="bar")
```
Comment (joeyorlando) [inline engine/apps/webhooks/tests/test_webhook.py:238]:
```suggestion
            assert expected_call.call_args == call("url", timeout=settings.OUTGOING_WEBHOOK_TIMEOUT, foo="bar")
```

File: engine/apps/webhooks/models/webhook.py [hunk 5]
```diff
@@ -246,17 +245,17 @@ def check_trigger(self, event_data):
...
+            r = requests.put(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "DELETE":
-            r = requests.delete(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.delete(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "OPTIONS":
-            r = requests.options(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.options(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "PATCH":
-            r = requests.patch(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.patch(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```
Comment (joeyorlando) [inline engine/apps/webhooks/models/webhook.py:258]:
```suggestion
            r = requests.patch(url, timeout=settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```

File: engine/apps/webhooks/models/webhook.py [hunk 6]
```diff
@@ -246,17 +245,17 @@ def check_trigger(self, event_data):
...
+            r = requests.post(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "PUT":
-            r = requests.put(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.put(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "DELETE":
-            r = requests.delete(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.delete(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "OPTIONS":
-            r = requests.options(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.options(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```
Comment (joeyorlando) [inline engine/apps/webhooks/models/webhook.py:256]:
```suggestion
            r = requests.options(url, timeout=settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```

File: engine/apps/webhooks/models/webhook.py [hunk 7]
```diff
@@ -246,17 +245,17 @@ def check_trigger(self, event_data):
...
+            r = requests.get(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "POST":
-            r = requests.post(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.post(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "PUT":
-            r = requests.put(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.put(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "DELETE":
-            r = requests.delete(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.delete(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```
Comment (joeyorlando) [inline engine/apps/webhooks/models/webhook.py:254]:
```suggestion
            r = requests.delete(url, timeout=settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```

File: engine/apps/webhooks/models/webhook.py [hunk 8]
```diff
@@ -246,17 +245,17 @@ def check_trigger(self, event_data):
...
     def make_request(self, url, request_kwargs):
         if self.http_method == "GET":
-            r = requests.get(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.get(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "POST":
-            r = requests.post(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.post(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "PUT":
-            r = requests.put(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.put(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```
Comment (joeyorlando) [inline engine/apps/webhooks/models/webhook.py:252]:
```suggestion
            r = requests.put(url, timeout=settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```

File: engine/apps/webhooks/models/webhook.py [hunk 9]
```diff
@@ -246,17 +245,17 @@ def check_trigger(self, event_data):
 
     def make_request(self, url, request_kwargs):
         if self.http_method == "GET":
-            r = requests.get(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.get(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
         elif self.http_method == "POST":
-            r = requests.post(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.post(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```
Comment (joeyorlando) [inline engine/apps/webhooks/models/webhook.py:250]:
```suggestion
            r = requests.post(url, timeout=settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```

File: engine/apps/webhooks/models/webhook.py [hunk 10]
```diff
@@ -246,17 +245,17 @@ def check_trigger(self, event_data):
 
     def make_request(self, url, request_kwargs):
         if self.http_method == "GET":
-            r = requests.get(url, timeout=OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
+            r = requests.get(url, timeout=live_settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```
Comment (joeyorlando) [inline engine/apps/webhooks/models/webhook.py:248]:
```suggestion
            r = requests.get(url, timeout=settings.OUTGOING_WEBHOOK_TIMEOUT, **request_kwargs)
```

File: engine/apps/webhooks/models/webhook.py [hunk 11]
```diff
@@ -12,9 +12,8 @@
 from django.utils import timezone
 from mirage import fields as mirage_fields
 from requests.auth import HTTPBasicAuth
-
+from apps.base.utils import live_settings
```
Comment (joeyorlando) [inline engine/apps/webhooks/models/webhook.py:15]:
```suggestion
from django.conf import settings
```

File: engine/apps/alerts/utils.py
Comment (joeyorlando) [inline engine/apps/alerts/utils.py:1]: you can leave this file as is, no need to modify 😄 

the `request_outgoing_webhook` method is deprecated and will be removed in a future release

File: engine/apps/webhooks/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joeyorlando) [reply engine/apps/webhooks/utils.py]: no problem! So for `engine/apps/alerts/utils.py` we can leave that untouched (that function is deprecated).

**Comment by joeyorlando** [comment]
@KevinDW-Fluxys do you mind adding this change to the "Added" section in the `CHANGELOG.md`? Thanks!

**Review by joeyorlando — COMMENTED** [review]
last two things:
- address the [linting issues](https://github.com/grafana/oncall/actions/runs/7741716976/job/21110045909?pr=3801#step:7:106)
- revert changes to `engine/apps/alerts/utils.py`

after that I'll go ahead and approve and merge this 😄

**Review by joeyorlando — APPROVED** [review]

**Comment by KevinDW-Fluxys** [comment]
@joeyorlando Missed a linting change. Thanks for the patient reviewing, I'm not so used to doing this, but its been fun :-)

**Pull request merged**